extendeddatasquare.go

```diff
@@ -15,29 +15,39 @@ import (
 type ExtendedDataSquare struct {
 	*dataSquare
 	codec             Codec
+	treeName          string
 	originalDataWidth uint
 }
 
 func (eds *ExtendedDataSquare) MarshalJSON() ([]byte, error) {
 	return json.Marshal(&struct {
 		DataSquare [][]byte `json:"data_square"`
 		Codec      string   `json:"codec"`
+		Tree       string   `json:"tree"`
 	}{
 		DataSquare: eds.dataSquare.Flattened(),
 		Codec:      eds.codec.Name(),
+		Tree:       eds.treeName,
 	})
 }
 
 func (eds *ExtendedDataSquare) UnmarshalJSON(b []byte) error {
 	var aux struct {
 		DataSquare [][]byte `json:"data_square"`
 		Codec      string   `json:"codec"`
+		Tree       string   `json:"tree"`
 	}
 
 	if err := json.Unmarshal(b, &aux); err != nil {
 		return err
 	}
-	importedEds, err := ImportExtendedDataSquare(aux.DataSquare, codecs[aux.Codec], NewDefaultTree)
+
+	treeConstructor, err := TreeFn(aux.Tree)
+	if err != nil {
+		return err
+	}
```

Comment:
Codecov has flagged several lines in this PR, including these, for insufficient test coverage. I believe it's sensible to create tests for these lines, but I'm curious whether this is a standard practice for this repo or not. @rootulp

Reply:
In general I think we want as much test coverage as possible but we're not strict about it. Put another way, it doesn't look like the CodeCov patch check is a required check prior to merge.

Reply:
I see, thanks for clarification! then I'd suggest developing tests to cover these lines and similar ones as much as possible.

Reply:
i added a test for unmarshalJSON to cover the default tree fallback logic